### ISSUE TITLE
orWhereNotNull but not andWhereNotNull

### ISSUE THREAD

**Issue by danschumann** [tgriesser/knex#746]
Why only allow for an `or` when talking about `notNull` methods?  Where's the `andWhereNotNull` ?

**Comment by bendrucker** [comment]
There's `andWhere` but having `and` methods for every single other where varient would get verbose.

**Comment by danschumann** [comment]
yea.. but I would think the same is true for `orWhere`.  What is the not null syntax again, `.andWhere('column', ..., null)`.   It's not `'not'`, because I tried that.  Anyway,  I don't see why `or` is included but `and` is not.  Either both or neither imo.

**Comment by bendrucker** [comment]
`or*` means something distinct whereas `and*` is always just an alias

**Comment by danschumann** [comment]
Internally, you mean?  I guess I'm still not seeing the difference, as in mysql
```select * from table where ... and ...```
and
```select * from table where ... or ...```
are similar orders of magnitude.

Anyway,  what is the current syntax then? `.andWhere('col', 'not', null)`?

It would be nice if `.where('col', null)` and `.where('col', 'not', null)` were in the docs, assuming it's implied that you must use `andWhere` for this.

**Comment by tgriesser** [comment]
Fun fact, if you want, you can actually just chain `.and.whereNotNull`. Basically if you call `whereNotNull` more than once it's an implied `and`

Really `andWhere` should just be deprecated as it's asymmetric with the rest of the API

**Comment by bendrucker** [comment]
What Tim said about deprecating `andWhere` is what I'm getting at. It's unnecessary grammar. It doesn't do anything. Whereas the `or` prefix actually means something.

**Comment by wubzz** [comment]
Despite the comments above this is way outdated and the arguments for removing or deprecating existing aliases are too weak in my opinion. Removing aliases now would just result in a breaking change, eventhough the aliases not having been documented.

It makes a lot more sense to keep supporting syntactic sugar when applicable. In the case of `andwhereNotNull`, a simple [one-line](https://github.com/tgriesser/knex/blob/master/src/query/builder.js#L985) contribution would do the trick:

`Builder.prototype.andWhereNotNull = Builder.prototype.whereNotNull`

**Comment by melissanoelle** [comment]
Ok, I'm new to knex, so the the confusion for me stems from `where()` and `andWhere()`. It _seems_ as if I should do something like `knex.raw(...).where(...).andWhere(...).andWhere(...)` etc, so when I also needed to check for null values my initial thought was that maybe I needed to do something like `knex.raw(...).whereNotNull(...).andWhere(...).andWhere(...)` but then if I needed to check for `null` more than once, what happens then? Since it _seemed_ as if I needed something syntactically sweetened with `and-`.

So, if it doesn't matter if I use `where()` or `andWhere()` and I can do something like this: `knex.raw(...).where(...).where(...).where(...)` or `knex.raw(...).where(...).whereNotNull(...).andWhere(...)` then .... everything is fine?

For my usecase, I ended up just using `knex.raw(...., [...])`, so I'm no longer stuck on this.
